docs/dev_guide/migration/debug-with-codepin.rst

```diff
@@ -0,0 +1,224 @@
+CodePin
+===============
+
+There are cases that the migrated SYCL program has different runtime behavior to
+the original CUDA program. The reason of the inconsistency could be:
+* Difference in arithmetic precision between hardware.
+* Semantic difference between the CUDA and SYCL APIs
+* Errors introduced during the automatic migration
+
+CodePin is introduced as a sub-feature of |tool_name| to reduce the effort of
+debugging such inconsist runtime behavior. When CodePin is enabled, |tool_name|
+will not only migrate the CUDA program to SYCL but also generate instrumented
+CUDA program.
+
+The instrumented code will dump the data of related variables before/after
```

Comment:
suggest changing to "The instrumented code will dump the data of related variables, before/after selected API or kernel calls, into a report."